### PR TITLE
Fix #4572 Add Store Unit Tests

### LINKED ISSUE #4572: Add unit tests for wallet stores
We have a number of Store type classes that handle integration between UI and brave-core controllers, these stores are currently untested but could possibly be tested via the test/mock controllers we use for SwiftUI previews.

A number of the test controller implementations need to be fleshed out to mimic brave-core functionality but overall I believe these store classes can be unit tested 

Security has mentioned that these two stores are most important for testing:
- `SendTokenStore`
- `SwapTokenStore`

### PR DESCRIPTION
**Pull request by nuo-xu**
This pull request fixes #4572 

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`

## Reviewer Checklist:

- [ ] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [ ] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [ ] Adequate unit test coverage exists to prevent regressions.
- [ ] Adequate test plan exists for QA to validate (if applicable).
- [ ] Issue and pull request is assigned to a milestone (should happen at merge time).

### CODE REVIEW

File: BraveWallet/Crypto/Stores/SendTokenStore.swift [hunk 1]
```diff
@@ -256,6 +256,13 @@ public class SendTokenStore: ObservableObject {
       }
     }
   }
+  
+  #if DEBUG
```
Comment (soner-yuksel) [inline BraveWallet/Crypto/Stores/SendTokenStore.swift:260]: Why do we have the DEBUG check here? We dont want tests to run except DEBUG or we dont want addresses to setup test addresses on this scheme?

File: BraveWallet/Preview Content/TestEthRpcController.swift [hunk 2]
```diff
@@ -33,7 +33,7 @@ class TestEthJsonRpcController: NSObject, BraveWalletEthJsonRpcController {
   }
   
   func balance(_ address: String, completion: @escaping (Bool, String) -> Void) {
-    completion(true, "10")
+    completion(true, "0x13e25e19dc20ba7")
```
Comment (soner-yuksel) [inline BraveWallet/Preview Content/TestEthRpcController.swift:37]: Can you comment what why these values are used in this completion block ?

**Review by soner-yuksel — APPROVED** [review]
:tada:

File: BraveWallet/Crypto/Stores/SendTokenStore.swift (hunk 1 above)
Comment (nuo-xu) [reply BraveWallet/Crypto/Stores/SendTokenStore.swift:260]: only in DEBUG scheme those private properties can be manipulated using this public method.

File: BraveWallet/Preview Content/TestEthRpcController.swift (hunk 2 above)
Comment (nuo-xu) [reply BraveWallet/Preview Content/TestEthRpcController.swift:37]: resolved at [c12c8d7](https://github.com/brave/brave-ios/pull/4741/commits/c12c8d7fae8c891efc4b8db300e7e651cea7c092)

**Pull request merged**